src/engine/GroupBy.h

```diff
@@ -298,10 +322,10 @@ class GroupBy : public Operation {
 
     // Returns a vector containing the offsets for all ids of `ids`,
     // inserting entries if necessary.
-    std::vector<size_t> getHashEntries(std::span<const Id> ids);
+    std::vector<size_t> getHashEntries(T<std::span<const Id>> ids);
 
     // Return the index of `id`.
-    [[nodiscard]] size_t getIndex(Id id) const { return map_.at(id); }
+    [[nodiscard]] size_t getIndex(T<Id> ids) const { return map_.at(ids); }
```

Comment:
There is a question, whether these should be passed by `const &`, but maybe that doesn't even matter (maybe a point for benchmarking).